### PR TITLE
feat: Adds support to ingest data via file monitor

### PR DESCRIPTION
**Pull request by rajkarkar-crest**
- Added ingestor to ingest data via file monitor using universal forwarder.

### CODE REVIEW

**Review by rfaircloth-splunk — APPROVED** [review]

File: docs/sample_generator.rst [hunk 1]
```diff
@@ -52,10 +52,14 @@ host_type = plugin | event
...
 
-input_type = modinput | scripted_input | syslog_tcp | file_monitor | windows_input | default
+input_type = modinput | scripted_input | syslog_tcp | file_monitor | windows_input | uf_file_monitor | default
     * The input_type used in addon to ingest data of a sourcetype used in stanza.
     * The way with which the sample data is ingested in Splunk depends on Splunk. The most similar ingesting approach is used for each input_type to get accurate index-time testing.
+    * In input_type=uf_file_monitor, universal forwarder will use file monitor to read event and then it will send data to indexer.
     * For example, in an Add-on, a sourcetype "alert" is ingested through syslog in live environment, provide input_type=syslog_tcp.
+    
+    .. warning::
+        uf_file_monitor input_type will only work with splunk_type=docker.
```
Comment (rajkarkar-crest) [inline docs/sample_generator.rst:62]: splunk-type

File: pytest_splunk_addon/standard_lib/event_ingestors/file_monitor_ingestor.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    """
+    def __init__(self, required_configs):
+        self.uf_host = required_configs.get("uf_host")
+        self.uf_port = required_configs.get("uf_port")
+        self.uf_username = required_configs.get("uf_username")
+        self.uf_password = required_configs.get("uf_password")
+        # Container of universal forwarder is linked with splunk instance.
+        # So using splunk_host as splunk and port 9997 directly.
+        self.splunk_host = "splunk"
+        self.splunk_s2s_port = "9997"
```
Comment (rajkarkar-crest) [inline pytest_splunk_addon/standard_lib/event_ingestors/file_monitor_ingestor.py:38]: Add parameter

File: pytest_splunk_addon/standard_lib/event_ingestors/ingestor_helper.py [hunk 3]
```diff
@@ -51,6 +53,8 @@ def ingest_events(cls, ingest_meta_data, addon_path, config_path, thread_count,
             input_type = event.metadata.get("input_type")
             if input_type in ["modinput", "windows_input", "syslog_tcp", "syslog_udp"]:
```
Comment (rajkarkar-crest) [inline pytest_splunk_addon/standard_lib/event_ingestors/ingestor_helper.py:54]: Add file monitor here

**Review by jayjoshi-crest — APPROVED** [review]

**Review by bhargav-nariyani-crest — APPROVED** [review]

**Comment by rajkarkar-crest** [comment]
I have read the CLA Document and I hereby sign the CLA

**Pull request merged**